arango/batch.py

```diff
@@ -102,12 +122,22 @@ def commit(self):
         for later retrieval via its :func:`arango.batch.BatchJob.result`
         method
 
+        :returns: list of arango.batch.BatchJob or None
+
         :raises arango.exceptions.BatchExecuteError: if the batch request
             cannot be executed
         """
+
+        res = self._lock.acquire(timeout=self._batch_submit_timeout)
```

Comment:
Maybe we should rename `batch_submit_timeout` to `lock_acquire_timeout` for clarity.

Reply:
But its not a timeout for the lock being acquired, its a timeout for the lock being acquired ONLY when submitting.  The end user doesn't care about the lock, they care that its the amount of time it would take until their batch either starts to submit or raises an error

Reply:
Sure that makes sense.